### ISSUE TITLE
Compliation for ARM fails

### ISSUE THREAD

**Issue by NCrashed** [erikd/wide-word#36]
I tried to use the library for our mobile app and getting the following errors at build time for `androideabi-ghc-8.6.5` platform:

```
[1 of 3] Compiling Data.WideWord.Word128 ( src/Data/WideWord/Word128.hs, dist/build/Data/WideWord/Word128.o )

src/Data/WideWord/Word128.hs:173:79: error:
    • Couldn't match expected type ‘ghc-prim-0.5.3:GHC.Prim.Word64#’
                  with actual type ‘ghc-prim-0.5.3:GHC.Prim.Word#’
    • In the first argument of ‘W64#’, namely ‘0##’
      In the first argument of ‘Word128’, namely ‘(W64# 0##)’
      In the expression: Word128 (W64# 0##) (W64# (int2Word# i#))
    |
173 | "fromIntegral :: Int -> Word128"     fromIntegral = \(I# i#) -> Word128 (W64# 0##) (W64# (int2Word# i#))
    |                                                                               ^^^

src/Data/WideWord/Word128.hs:173:91: error:
    • Couldn't match expected type ‘ghc-prim-0.5.3:GHC.Prim.Word64#’
                  with actual type ‘ghc-prim-0.5.3:GHC.Prim.Word#’
    • In the first argument of ‘W64#’, namely ‘(int2Word# i#)’
      In the second argument of ‘Word128’, namely ‘(W64# (int2Word# i#))’
      In the expression: Word128 (W64# 0##) (W64# (int2Word# i#))
    |
173 | "fromIntegral :: Int -> Word128"     fromIntegral = \(I# i#) -> Word128 (W64# 0##) (W64# (int2Word# i#))
    |                                                                                           ^^^^^^^^^^^^
```

Here the full log of build and full toolchain:
https://travis-ci.com/hexresearch/ergvein/jobs/288072584#L5513

P.S. Sorry that I don't have a concise example of the fail.

**Comment by erikd** [comment]
Please try the fix above.

**Comment by NCrashed** [comment]
> Please try the fix in PR #37 .

It was not enough to fix all errors.
